db/keyval/bolt/bytes_broker_impl.go

```diff
@@ -0,0 +1,684 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package bolt
+
+import (
+	"fmt"
+	"github.com/boltdb/bolt"
+	"github.com/ligato/cn-infra/datasync"
+	"github.com/ligato/cn-infra/db/keyval"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/logrus"
+	"os"
+	"strings"
+)
+
+var boltLogger = logrus.NewLogger("bolt")
+
+func init() {
+	if os.Getenv("DEBUG_BOLT_CLIENT") != "" {
+		boltLogger.SetLevel(logging.DebugLevel)
+	}
+}
+
+// Client serves as a client for Bolt KV storage and implements keyval.CoreBrokerWatcher interface.
+type Client struct {
+	dbPath          *bolt.DB
+	bucketSeparator string
+}
+
+func transformKey(key string, separator string) ([][]byte, []byte) {
+	if !strings.HasPrefix(key, separator) {
+		return nil, nil
+	}
+
+	names := strings.Split(key, separator)
+
+	var bucketNames [][]byte // := names[:len(names)-1]
+	for _, name := range names[1 : len(names)-1] {
+		bucketNames = append(bucketNames, ([]byte)(name))
+	}
+	keyInBucket := ([]byte)(names[len(names)-1])
+
+	return bucketNames, keyInBucket
+}
+
+// NewTxn creates new transaction
+func (client *Client) NewTxn() keyval.BytesTxn {
+	tx, _ := client.dbPath.Begin(true)
+	return &txn{
+		readonly:  false,
+		separator: client.bucketSeparator,
+		kv:        tx,
+	}
+}
+
+// Create bucket base on given names in current transaction
+func createBucket(tx *bolt.Tx, bucketNames [][]byte) (*bolt.Bucket, error) {
+	var bucket *bolt.Bucket
+	var bucketName []byte
```

Comment:
You don't need to declare all variables, especially not if used in single scope such as the for loop below. You can use `for _, bucketName := range bucketNames`.